### PR TITLE
[log-shipper] New socket destination

### PR DESCRIPTION
**Pull request by nabokihms**
## Description
A new type, `Socket`, of cluster log destination has been introduced. Now log-shipper agents can use TCP or UDP sockets to send logs and format them as they wish. Provided formatting options are:
* text
* JSON
* CEF
* Syslog

## Why do we need it, and what problem does it solve?
Related to https://github.com/deckhouse/deckhouse/issues/8700

## Checklist
- [x] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: log-shipper
type: feature
summary: |
  Add an ability to send logs via socket (TCP/UDP). 
  Messages can be formatted as text, JSON, CEF, or Syslog. 
```

### CODE REVIEW

**Comment by nabokihms** [comment]
Tested it using vector as a syslog server:

`root@standard-master-0:~# cat syslog.yaml`
```yaml
apiVersion: deckhouse.io/v1alpha1
kind: ClusterLogDestination
metadata:
  name: syslog
spec:
  type: Socket
  socket:
    address: vector-headless.vector:9000
    encoding:
      codec: Syslog
    mode: TCP
```

Result:

```
root@standard-master-0:~# kubectl -n vector logs vector-0 --tail=5
{"facility":"user","host":"standard-worker-127674b8-8676d-2ldgj","hostname":"standard-worker-127674b8-8676d-2ldgj","message":"nc: bad address 'memcached:11211'","severity":"info","source_ip":"10.111.1.180","source_type":"syslog","timestamp":"2024-07-01T13:19:36.444954907Z","version":1}
{"facility":"user","host":"standard-master-0","hostname":"standard-master-0","message":"E0622 23:03:19.280782       1 utils.go:95] [ID:232900] GRPC error: rpc error: code = Unimplemented desc = GetCapacity is not yet implemented","severity":"info","source_ip":"10.111.0.93","source_type":"syslog","timestamp":"2024-06-22T23:03:19.280903590Z","version":1}
{"facility":"user","host":"standard-master-0","hostname":"standard-master-0","message":"E0622 23:03:19.338021       1 utils.go:95] [ID:232901] GRPC error: rpc error: code = Unimplemented desc = GetCapacity is not yet implemented","severity":"info","source_ip":"10.111.0.93","source_type":"syslog","timestamp":"2024-06-22T23:03:19.338099941Z","version":1}
```

File: modules/460-log-shipper/crds/doc-ru-cluster-log-destination.yaml [hunk 1]
```diff
@@ -229,6 +229,41 @@ spec:
...
+                socket:
+                  properties:
+                    address:
+                      description: "Адрес сокета."
+                    encoding:
+                      description: "В каком формате закодировать сообщение."
+                    mode:
+                      descriptions: "Тип сокета, который будет использован."
+                    tcp:
+                      descriptions: "Параметры для настройки TCP сокетов."
```
Comment (z9r5) [inline modules/460-log-shipper/crds/doc-ru-cluster-log-destination.yaml:241]:
```suggestion
                      descriptions: "Параметры для настройки TCP-сокетов."
```

**Review by z9r5 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

File: modules/460-log-shipper/crds/doc-ru-cluster-log-destination.yaml [hunk 2]
```diff
@@ -229,6 +229,41 @@ spec:
...
+                                  description: |
+                                    Закодированный в Base64 ключ в формате PEM.
+
+                                    Также необходимо указать сертификат в параметре `crtFile`.
+                                keyPass:
+                                  description: Закодированный в Base64 пароль для ключа.
+                        verifyHostname:
+                          description: Проверка соответствия имени удаленного хоста и имени, указанного в TLS-сертификате удаленного хоста.
+                        verifyCertificate:
+                          description: Проверка действия TLS-сертификата удаленного хоста.
```
Comment (z9r5) [inline modules/460-log-shipper/crds/doc-ru-cluster-log-destination.yaml:266]:
```suggestion
                          description: |
                            Проверка действия TLS-сертификата удаленного хоста.
                            
                            Если установлено значение `false`, сертификат не проверяется в списках отзыва сертификатов.
```

**Review by z9r5 — APPROVED** [review]

File: modules/460-log-shipper/docs/EXAMPLES.md [hunk 3]
```diff
@@ -304,6 +304,28 @@ spec:
...
+  type: Socket
+  socket:
+    mode: TCP
+    address: 192.168.0.1:3000
+    encoding: 
+      codec: Syslog
+  extraLabels:
+    syslog.severity: "alert"
+    # the request_id field should be present in the log message
+    syslog.message_id: "{{ request_id }}"
```
Comment (yalosev) [inline modules/460-log-shipper/docs/EXAMPLES.md:326]: Don't we need some validation/defaults for this?

**Review by yalosev — APPROVED** [review]

File: modules/460-log-shipper/docs/EXAMPLES.md (hunk 3 above)
Comment (nabokihms) [reply modules/460-log-shipper/docs/EXAMPLES.md:326]: There is a default '-'

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [8905](https://github.com/deckhouse/deckhouse/pull/8905) to the branch [release-1.62](https://github.com/deckhouse/deckhouse/tree/release-1.62) successful!

**Pull request merged**